### DISCUSSION TITLE
TapTap distribution

### DISCUSSION THREAD

**Discussion by Hexality** [ppy/osu#20471] [Q&A]
Hi, I always wondered this so I came here to ask
Did you ever upload osu!lazer to taptap? I saw osu!lazer there distributed under **ppy Pty Ltd** and started to wonder if you were the one to upload it to the platform.
[osu!Lazer on Taptap.io](https://www.taptap.io/app/170963?share_id=6297a5e06cb1&utm_medium=share&utm_source=copylink)

The build in there is quite old and probably unsupported as its apparently the version that got uploaded to play store in the past.

![image](https://user-images.githubusercontent.com/17398632/192162403-94a62a10-aa6d-4a51-a09c-3f89326170ff.png)

I've been redirecting people there to the [latest releases](https://github.com/ppy/osu/releases/latest) page since 2019 but never thought about asking this here before.

>> Version specified in the page info:
>>> 
![image](https://user-images.githubusercontent.com/17398632/192162495-e3a50c24-6cb1-484d-91b1-4d902fd50a42.png)

**Comment by agatemosu** [comment]
I suppose that when it is in the Google Play Store it will update automatically. If not, there is a channel on [Discord's TapTap server](https://discord.gg/taptap) called `#🔶⎮update-game-on-taptap`.

**Comment by WindowsMeosu** [comment]
doing a quick search, search engines don't mention anything by peppy nor any osu! developer uploading lazer to taptap. If it was uploaded by peppy, I'd assume it would be at least up to date most of the time. The fact that it was even uploaded there itself is a bit skeptical.

**Reply by agatemosu** [reply]
TapTap has a [Official FAQ for Developers](https://www.taptap.io/post/861092), and I think the second question is what happened.

**Reply by Hexality** [reply]
considering osu!lazer was once uploaded to playstore in beta testing mode, maybe taptap scraped it just like they do with other games idk.
